sam/record.go

```diff
@@ -255,7 +255,7 @@ func (r *Record) UnmarshalSAM(h *Header, b []byte) error {
 	}
 	if !bytes.Equal(f[9], []byte{'*'}) {
 		r.Seq = NewSeq(f[9])
-		if !r.Cigar.IsValid(r.Seq.Length) {
+		if !(len(r.Cigar) == 0 || r.Cigar.IsValid(r.Seq.Length)) {
```

Comment:
When I read the negation distributed over the conjunction I need to understand both parts of the conjunction and then negate. With the disjunction I can understand as I read.